### PR TITLE
Readonly logic fixed for ROX&RWO mount

### PR DESCRIPTION
**Pull request by EnvyusKennys**
Tests edited for testing readonly conditions
host accessmode change WIP

### CODE REVIEW

**Comment by EnvyusKennys** [comment]
For ceph:
In order to allow ROX&RWO mount, user need to append rbac for both clusterroles rbd-external-provisioner-runner and cephfs-external-provisioner-runner with [get,list,watch] for pods and csinodes.storage.k8s.io

**Comment by EnvyusKennys** [comment]
@houz42

File: pkg/controller/csi_handler.go [hunk 1]
```diff
@@ -550,8 +550,9 @@ func (h *csiHandler) csiAttach(va *storage.VolumeAttachment) (*storage.VolumeAtt
 	if err != nil {
 		return va, nil, err
 	}
-
-	publishInfo[readonlyAttachmentKey] = strconv.FormatBool(readOnly)
+	if _, ok := publishInfo[readonlyAttachmentKey]; ok {
```
Comment (houz42) [inline pkg/controller/csi_handler.go]: ok 表示 publisInfo map 有 readonlyAttachmentKey 这个 key. 此时将这个 key 的值 set 为 readOnly. 那如果 map 里还没有这个 key, 就不写 readOnly 了吗?

Comment (EnvyusKennys) [reply pkg/controller/csi_handler.go]: 重复判断了 昨天最后改的时候csi那边已经把attachment key的值设置好通过publishinfo传回来了这边只需要判断是否map里有attachment key即可

Comment (houz42) [reply pkg/controller/csi_handler.go]: 那修改一下

**Review by houz42 — APPROVED** [review]
LGTM

**Pull request merged**